repository/repository-core/src/test/java/org/eclipse/vorto/repository/core/RequestRepositorySessionHelperTest.java

```diff
@@ -0,0 +1,347 @@
+ * https://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.vorto.repository.core;
+
+import org.eclipse.vorto.repository.AbstractIntegrationTest;
+import org.eclipse.vorto.repository.core.impl.RequestRepositorySessionHelper;
+import org.junit.Assert;
+import org.junit.Test;
+import org.xml.sax.ContentHandler;
+import org.xml.sax.SAXException;
+
+import javax.jcr.*;
+import javax.jcr.lock.LockException;
+import javax.jcr.nodetype.ConstraintViolationException;
+import javax.jcr.nodetype.NoSuchNodeTypeException;
+import javax.jcr.retention.RetentionManager;
+import javax.jcr.security.AccessControlManager;
+import javax.jcr.version.VersionException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.security.AccessControlException;
+
+public class RequestRepositorySessionHelperTest extends AbstractIntegrationTest {
+
+    private static String TEST_TENANT_ID = "TEST_TENANT_ID";
+
+    @Test
+    public void testInternalSessionSupplierWired() {
+        // create a session helper with autowiring (live http request)
+        RequestRepositorySessionHelper helper = new RequestRepositorySessionHelper();
+        helper.afterPropertiesSet();
+        helper.setTenantId(TEST_TENANT_ID);
+        helper.setRepository(createMockRepository());
+        Session mySession = helper.getSession();
+        Assert.assertNotNull(mySession);
+        Assert.assertTrue(mySession.isLive());
+        // session should be kept alive for potential reuse
+        helper.logoutSessionIfNotReusable(mySession);
+        Assert.assertTrue(mySession.isLive());
+        // same session should be returned when requested again
+        Session sessionForReuse = helper.getSession();
+        Assert.assertTrue(mySession.equals(sessionForReuse));
+        // after destroy session should be logged out
+        helper.destroy();
+        Assert.assertFalse(sessionForReuse.isLive());
+    }
+
+    @Test
+    public void testInternalSessionSupplierNonWired() {
+        // create a session helper without autowiring (no http request)
+        RequestRepositorySessionHelper helper = new RequestRepositorySessionHelper(false);
+        helper.setTenantId(TEST_TENANT_ID);
+        helper.setRepository(createMockRepository());
+        Session mySession = helper.getSession();
+        Assert.assertNotNull(mySession);
+        Assert.assertTrue(mySession.isLive());
+        // session should be logged out - no reuse
+        helper.logoutSessionIfNotReusable(mySession);
+        Assert.assertFalse(mySession.isLive());
+        // new session should be returned
+        Session sessionForReuse = helper.getSession();
+        Assert.assertFalse(mySession.equals(sessionForReuse));
+        // after destroy session should still be alive
+        helper.destroy();
+        Assert.assertTrue(sessionForReuse.isLive());
+
+    }
+
+    private Repository createMockRepository() {
```

Comment:
Wouldn't it be easier to use Mockito here? Or is there a reason not to use it for this test?

Reply:
Hi @kolotu ,
thanks. I will adapt this.

Reply:
Can't check out the branch right now but I agree with the mock-related comment. Also (and forgive me if that's obvious) - are the invocation to `get` on the supplier always guaranteed to not return `null`?

Reply:
Hi @mena-bosch ,
I didn't change the behaviour compared to the way it was before.